Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -1259,6 +1259,27 @@ void NetworkStorageManager::syncLocalStorage(CompletionHandler<void()>&& complet
     });
 }
 
+void NetworkStorageManager::fetchLocalStorage(CompletionHandler<void(HashMap<WebCore::ClientOrigin, HashMap<String, String>>&&)>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+    ASSERT(!m_closed);
+
+    protectedWorkQueue()->dispatch([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)]() mutable {
+        assertIsCurrent(workQueue());
+
+        HashMap<WebCore::ClientOrigin, HashMap<String, String>> localStorageMap;
+
+        for (auto& [clientOrigin, manager] : m_originStorageManagers) {
```

Comment:
`m_originStorageManager` only tracks the origins that are active. Technically we should use `getAllOrigins()` for all data.